waiter/src/waiter/util/http_utils.clj

```diff
@@ -76,9 +80,52 @@
         (.setUserAgentField client new-user-agent-field)))
     client))
 
+(defn- prepare-http2-transport
+  "Returns the HTTP/2 client transport."
+  [connection-timeout-ms]
+  (let [http2-client (HTTP2Client.)
+        http2-protocols (ArrayList. ["h2" "h2c"])]
+    (.setConnectTimeout http2-client connection-timeout-ms)
+    (.setProtocols http2-client http2-protocols)
+    (HttpClientTransportOverHTTP2. http2-client)))
+
+(defn prepare-http-clients
+  "Prepares and returns a map of HTTP clients for http/1 and http/2 requests."
+  [{:keys [conn-timeout user-agent] :as config}]
+  (let [http2-transport (prepare-http2-transport conn-timeout)]
+    {:http1-client (http-client-factory (cond-> config
+                                          user-agent (update :user-agent str ".http1")))
+     :http2-client (-> (cond-> config
+                         user-agent (update :user-agent str ".http2"))
+                       (assoc :transport http2-transport)
+                       http-client-factory)}))
+
+(defn retrieve-http-client
+  "Returns the appropriate http client based on the backend protocol."
+  [backend-proto {:keys [http1-client http2-client]}]
+  (if (contains? #{"h2" "h2c"} backend-proto)
```

Comment:
Should we validate that we are otherwise receiving `http` or `https`?  Log an error if not?